### PR TITLE
Enhancement: Supports adding third-party dependencies from git repo with the version field

### PR DESCRIPTION
**Pull request by Manoramsharma**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

fix : #266 

#### 2. What is the scope of this PR (e.g. component or file name):
pkg/client/client.go

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [x] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

I simply changed the logic of AddDepToPkg function to handle version field at the time of adding the dependencies to the pkg. 
#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

I directly modified the test_data for testing the unit tests. No need to make any other changes in unit test.

image

### CODE REVIEW

**Comment by Manoramsharma** [comment]
Hi @zong-zhe 
Kindly review this one , I have checked on the failing tests and its not because of my changes. Thanks 😄

File: pkg/client/client.go [hunk 1]
```diff
@@ -829,6 +829,12 @@ func (c *KpmClient) AddDepToPkg(kclPkg *pkg.KclPkg, d *pkg.Dependency) error {
 		return err
 	}
 
+	dep, err := c.LoadPkgFromPath(c.getDepStorePath(c.homePath, d, false))
+	if err != nil {
+		return fmt.Errorf("failed to load downloaded package: %w", err)
+	}
+	d.Source.Git.Version = dep.GetPkgVersion()
```
Comment (zong-zhe) [inline pkg/client/client.go:836]: Here might be some other cases, not just git, if the dependency is not from git repo, `d.Source.Git` will be `nil`.

File: pkg/client/test_data/add_with_git_commit/test_pkg/kcl.mod.expect [hunk 2]
```diff
@@ -4,4 +4,4 @@ edition = "0.0.1"
 version = "0.0.1"
 
 [dependencies]
-flask-demo-kcl-manifests = { git = "https://github.com/kcl-lang/flask-demo-kcl-manifests.git", commit = "ade147b" }
+flask-demo-kcl-manifests = { git = "https://github.com/kcl-lang/flask-demo-kcl-manifests.git", commit = "ade147b", version = "0.0.1" }
```
Comment (zong-zhe) [inline pkg/client/test_data/add_with_git_commit/test_pkg/kcl.mod.expect:7]: Add more test cases for

1. Add dependencies from git repo by git commit, branch and tag, to test whether the `version` is updated to kcl.mod.

2. Add more test case for if the git dependency with `version` field in kcl.mod, `update` or `run` should check whether  version exists in the current commit, branch or tag.

**Comment by Manoramsharma** [comment]
@Peefy Can you please review this one. I have added the required changes.

File: pkg/client/test_data/add_with_git_commit/test_pkg/kcl.mod.expect (hunk 2 above)
Comment (Manoramsharma) [reply pkg/client/test_data/add_with_git_commit/test_pkg/kcl.mod.expect:7]: @zong-zhe I have added the the tests, Could you please review now!!!!

**Pull request closed without merging**